src/types/signed.rs

```diff
@@ -0,0 +1,55 @@
+use crate::types::{Address, Bytes, H256, U256};
+use serde::{Deserialize, Serialize};
+
+/// Struct representing signed data returned from `Accounts::sign` method.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct SignedData {
+    /// The original method that was signed.
+    pub message: String,
+    /// The keccak256 hash of the signed data.
+    #[serde(rename = "messageHash")]
+    pub message_hash: H256,
+    /// V value.
+    pub v: u8,
+    /// R value.
+    pub r: H256,
+    /// S value.
+    pub s: H256,
+    /// The signature bytes.
+    pub signature: Bytes,
+}
+
+/// Transaction data for signing. The `Accounts::sign_transaction` method will
+/// fill optional fields with sane defaults when they are ommited.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct TransactionParameters {
+    /// Transaction nonce (None for account transaction count)
+    pub nonce: Option<U256>,
+    /// To address
+    pub to: Option<Address>,
+    /// Supplied gas (None for sensible default)
+    pub gas: Option<U256>,
+    /// Gas price (None for sensible default)
+    pub gas_price: Option<U256>,
+    /// Transfered value (None for no transfer)
```

Comment:
Why `None` and not `H256::zero()` or `Default::default()`? I think it makes it really confusing if there is any difference between `None` and `Some(0)` (afaik there isn't).

Reply:
For gas price specifically there is a difference. `Some(0)` means that you want to pay `0` for gas (transaction will never be mined though). `None` on the other hand will tell the signing function to get the current recommended gas price and use that.

Reply:
Yes, I only meant `value` and `data` field. The rest is fine.